### PR TITLE
Transactions - Use Pill component to display risk levels in transactions list

### LINKED ISSUE #6038: Make RiskLevel use the Pill component for UI consistency
### Describe the bug
Currently the RiskLevel that renders the risk levels in the transactions list, ( `Payments > Transactions` ) seems to be using colors on plain text based on levels. 

image

For UI consistency, it was suggested to use the standard WooCommerce Pill component instead. 

image

### To Reproduce

1. Checkout this branch and do a few transactions using normal and Risky (fraud testing ) test cards 
2. Browse to `Payments > Transactions` 
3. Unhide the `Risk Level` column, if its hidden.
4. Observe how the risk level appears ( whether as colored plain text or as pill ) 
5. Repeat the steps with Authorize and capture transactions. 

### Actual behavior
The risk level currently appears as plain text in colors depending on the risk level.

### Expected behavior
The risk levels should render as colored pills

### Additional context
Discussed here : pdjVC1-1vj-p2#comment-2153

### PR DESCRIPTION
**Pull request by nagpai**
Fixes #6038 

#### Changes proposed in this Pull Request

Risk level column within `Payments > Transactions` to be displayed using the Pill component, instead of plain text with color. This is to make it consistent with UI guidelines of WC and WC Pay. 

##### Before 
image

##### After
image

#### Testing instructions

1. Checkout this branch and do a few transactions using normal and Risky (fraud testing ) test cards
2. Browse to `Payments > Transactions`
3. Unhide the Risk Level column, if its hidden.
4. The Risk level should be visible within a Pill component. 
5. Test the same steps with Authorize and capture transactions.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `6041` or branch name `update/6038-1-use-pill-for-risk-level` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&branches.woocommerce-payments=update/6038-1-use-pill-for-risk-level) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: afb4c7531c427fc80c0fe3247a6e58309ba0a683
- Build time: 2023-04-18 02:48:53 UTC

_Note: the build is updated when a new commit is pushed to this PR._

**Review by mgascam — APPROVED** [review]
LGTM! (I did code review only)

File: client/components/risk-level/index.js [hunk 1]
```diff
@@ -4,13 +4,14 @@
  * Internal dependencies
  */
 import riskMappings from './strings';
```
Comment (kalessil) [inline client/components/risk-level/index.js]: Can we turn this file into TypeScript, without touching it's consumers?

Comment (nagpai) [reply client/components/risk-level/index.js]: I have attempted this here 
7983cc8

Requesting advance forgiveness @kalessil  :)

File: client/components/risk-level/index.tsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+
+/**
+ * Internal dependencies
+ */
+import riskMappings from './strings';
+import Pill from '../pill';
+
+type RiskLevelNames = 'normal' | 'elevated' | 'highest' | 'not_assessed';
+
+type RiskColorMapping = 'success' | 'alert' | 'danger' | 'light';
```
Comment (kalessil) [inline client/components/risk-level/index.tsx]: RilskLevelColours perhaps?

File: client/components/risk-level/test/index.tsx [hunk 3]
```diff
@@ -3,14 +3,18 @@
...
 /**
  * Internal dependencies
  */
-import RiskLevel, { calculateRiskMapping } from '../';
+import RiskLevel, { calculateRiskMapping } from '..';
 
 describe( 'RiskLevel', () => {
+	function renderRisk( risk: number ): HTMLElement {
+		return render( <RiskLevel risk={ risk } /> ).container;
+	}
```
Comment (kalessil) [inline client/components/risk-level/test/index.tsx:17]: Can we add an empty line after the function, please?

File: client/components/risk-level/index.tsx [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import riskMappings from './strings';
+import Pill from '../pill';
+
+type RiskLevelNames = 'normal' | 'elevated' | 'highest' | 'not_assessed';
```
Comment (kalessil) [inline client/components/risk-level/index.tsx:14]: @Automattic/pulsar, any opinions on keeping the definitions here vs keeping them under types definitions in  'client/types/' folder?

**Review by kalessil — CHANGES_REQUESTED** [review]

File: client/components/risk-level/index.tsx (hunk 4 above)
Comment (nagpai) [reply client/components/risk-level/index.tsx:14]: I was divided here, but then saw other components having types in the same file, where number of types were fewer and with less complex interfaces. [Example](https://github.com/Automattic/woocommerce-payments//blob/b6e37ebf103aba3dc9be58daf9be647fcee14ce8/client/components/deposit-status-pill/index.tsx)

I will be happy to comply what we decide as the best practice. 👍

File: client/components/risk-level/test/index.tsx (hunk 3 above)
Comment (nagpai) [reply client/components/risk-level/test/index.tsx:17]: Sorry. Done here 16ebccd

File: client/components/risk-level/index.tsx (hunk 2 above)
Comment (nagpai) [reply client/components/risk-level/index.tsx]: That certainly sounds much better, Thanks. I have changed it here 16ebccd

**Review by kalessil — APPROVED** [review]
Approving!

**Comment by nagpai** [comment]
Tested with Oldest supported version - WC 7.3 and  the pills seem to render correctly. Thanks for the reminder @kalessil !

**Pull request merged**